From 86edd82a064269dd1417e89a834a7926c0fe4dd2 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:06:31 -0400
Subject: [powerpc] Add includes to fix powernv/rng.c build

Message-id: <1410545655-205645-163-git-send-email-dzickus@redhat.com>
Patchwork-id: 94170
O-Subject: [RHEL7 PATCH 162/626] powerpc: Add includes to fix powernv/rng.c build
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 1400b4204ca334bb67be5b091b45214b57912b33
Author: Stephen Rothwell <sfr@canb.auug.org.au>
Date:   Mon Oct 28 19:34:41 2013 +1100

    powerpc: Add includes to fix powernv/rng.c build

    Caused by commit a4da0d50b2a0 ("powerpc: Implement
    arch_get_random_long/int() for powernv") from the powerpc tree
    interacting with commit b5b4bb3f6a11 ("of: only include prom.h on sparc")
    from the dt-rh tree.

    I added this merge fix patch (which will need to be sent to Linus when
    these two trees get merged, or could be applied now to the powerpc tree):

    [ Also add linux/smp.h to get cpu_to_chip_id -- BenH ]

    Signed-off-by: Stephen Rothwell <sfr@canb.auug.org.au>
    Acked-by: Rob Herring <rob.herring@calxeda.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/powernv/rng.c b/arch/powerpc/platforms/powernv/rng.c
index 02db7d7..8844628 100644
--- a/arch/powerpc/platforms/powernv/rng.c
+++ b/arch/powerpc/platforms/powernv/rng.c
@@ -11,10 +11,13 @@
 
 #include <linux/kernel.h>
 #include <linux/of.h>
+#include <linux/of_address.h>
 #include <linux/of_platform.h>
 #include <linux/slab.h>
+#include <linux/smp.h>
 #include <asm/archrandom.h>
 #include <asm/io.h>
+#include <asm/prom.h>
 #include <asm/machdep.h>
 
 
-- 
1.7.1